bson/src/main/org/bson/codecs/configuration/mapper/ClassModel.java

```diff
@@ -34,6 +35,10 @@
     private final Map<String, FieldModel> fields = new TreeMap<String, FieldModel>();
     private final CodecRegistry registry;
     private final TypeResolver resolver;
+    private final MemberResolver memberResolver;
+    private final WeightedValue<String> collectionName = new WeightedValue<String>();
```

Comment:
We have to consider how this will be used in the context of the Java driver.  Currently, you have to be explicit about the collection name:

     database.getCollection("people", Person.class);

I suppose the collectionName could be ignored when used this way, and could be taken into account with an alternate API like:

    database.getCollection(Person.class)

For this usage, the Codec interface would have to be optionally extendable to provide the collection name.